internal/dev_server/ui/vite.config.ts

```diff
@@ -0,0 +1,26 @@
+import { PluginOption, defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import svgr from 'vite-plugin-svgr';
+import { viteSingleFile } from 'vite-plugin-singlefile';
+
+// https://vitejs.dev/config/
+export default defineConfig(({ command }) => {
+  const plugins: PluginOption[] = [react(), svgr()];
+
+  if (command === 'build') {
+    plugins.push(viteSingleFile());
+  }
+
+  return {
+    plugins,
+    server: {
+      proxy: {
+        '/api': {
+          target: 'http://localhost:8765',
```

Comment:
Is this gonna be confusing the user specifies a different port?

Reply:
This is actually just to make it easier to develop the UI, so configurability isn't needed here. When you spin up the vite server to iterate on the LaunchDevly UI, this config makes the UI server reverse proxy the API.